src/unix/fs.c

```diff
@@ -231,7 +231,45 @@ static ssize_t uv__fs_futime(uv_fs_t* req) {
 
 
 static ssize_t uv__fs_mkdtemp(uv_fs_t* req) {
+#if defined(__MVS__)
+  /* There is no mkdtemp. So instead use mktemp to generate a
```

Comment:
@sam-github , I think this is somewhat functionally equivalent.

1. mkstemp (supported on zos) to create a file with a randomly generated a random string
2. delete the file and then mkdir a directory with the same random name
3. If step 2. failed go back to step 1 and repeat until step 2 succeeds.

However, I added a TODO to copy the windows as you suggested. Will do that as soon as this PR goes through.
@saghul mentioned that this PR is already too large :-)